feature/system/gnmi/cliorigin/mixed_oc_cli_origin_support_test/mixed_oc_cli_origin_support_test.go

```diff
@@ -0,0 +1,217 @@
+// limitations under the License.
+
+// Package mixed_oc_cli_origin_support_test implements GNMI 1.12 from go/wbb:vendor-testplan
+package mixed_oc_cli_origin_support_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ygot/ygot"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func interfaceDescriptionCLI(dp *ondatra.Port, desc string) string {
+	switch dp.Device().Vendor() {
+	case ondatra.ARISTA, ondatra.CISCO:
+		const tmpl = `
+interface %s
+  description %s
+`
+		return fmt.Sprintf(tmpl, dp.Name(), desc)
+	}
+	return ""
+}
+
+func buildOCUpdate(path *gpb.Path, value string) *gpb.Update {
+	if len(path.GetElem()) == 0 || path.GetElem()[0].GetName() != "meta" {
+		path.Origin = "openconfig"
+	}
+	update := &gpb.Update{
+		Path: path,
+		Val:  &gpb.TypedValue{Value: &gpb.TypedValue_StringVal{StringVal: value}},
+	}
+	return update
+}
+
+func buildCLIUpdate(value string) *gpb.Update {
+	update := &gpb.Update{
+		Path: &gpb.Path{
+			Origin: "cli",
+			Elem:   []*gpb.PathElem{},
+		},
+		Val: &gpb.TypedValue{
+			Value: &gpb.TypedValue_AsciiVal{
+				AsciiVal: value,
+			},
+		},
+	}
+	return update
+}
+
+// Push overlapping mixed SetRequest specifying CLI before OpenConfig for  DUT port-1.
+func TestOrderDependenceForCli(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	dp := dut.Port(t, "port1")
+	// `origin: "cli"` - containing vendor  configuration.
+	// ~~~
+	// interface <DUT port-1>
+	// description foo1
+	//  ```
+	// ~~~
+	intfConfig := interfaceDescriptionCLI(dp, "want")
+	if intfConfig == "" {
+		t.Skip("Vendor is not supported.")
+	}
+	gnmiClient := dut.RawAPIs().GNMI().Default(t)
+	t.Logf("Building the CLI config:\n%s", intfConfig)
+
+	// `origin: ""` (openconfig, default origin) setting the DUT port-1
+	//  string value at `/interfaces/interface/config/description` to `"foo2"`.
+	resolvedPath := dut.Config().Interface(dp.Name()).Description()
+	path, _, errs := ygot.ResolvePath(resolvedPath)
+
+	gpbSetRequest := &gpb.SetRequest{
+		Update: []*gpb.Update{
+			buildOCUpdate(path, "not want"),
+			buildCLIUpdate(intfConfig),
+		},
+	}
+	if errs != nil {
+		t.Fatalf("Could not resolve path: %v", errs)
+	}
+	t.Log("gnmiClient Set both CLI and OpenConfig modelled config")
+	t.Log(gpbSetRequest)
+
+	response, err := gnmiClient.Set(context.Background(), gpbSetRequest)
+	if err != nil {
+		t.Fatalf("gnmiClient.Set() with unexpected error: %v", err)
+	}
+	t.Log("gnmiClient Set Response for CLI and OpenConfig modelled config")
+	t.Log(response)
+
+	// Validate that DUT port-1 description has `"foo2"`
+	got := dut.Telemetry().Interface(dp.Name()).Description().Get(t)
+	if got != "want" {
+		t.Errorf("Get(DUT port description): got %v, want %v", got, "want")
+	}
+}
+
+// Push overlapping mixed SetRequest specifying OpenConfig before Cli for  DUT port-1.
```

Comment:
This entire function seems to be using spaces instead of tabs. Please use gofmt to reformat.